warehouse/oidc/models/google.py

```diff
@@ -0,0 +1,96 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from sqlalchemy import Column, ForeignKey, String, UniqueConstraint
+from sqlalchemy.dialects.postgresql import UUID
+
+from warehouse.oidc.models._core import (
+    OIDCPublisher,
+    PendingOIDCPublisher,
+    _check_claim_binary,
+    _check_claim_invariant,
+)
+
+
+class GooglePublisherMixin:
+    """
+    Common functionality for both pending and concrete Google OIDC
+    providers.
+    """
+
+    email = Column(String, nullable=False)
+    sub = Column(String, nullable=False)
```

Comment:
I'm thinking this might be `nullable`: https://github.com/pypi/warehouse/issues/13551#issuecomment-1531959766

Reply:
Sounds good! Will tweak.